Extensions: Add Advanced Tab settings for WP Super Cache

donnapep opened the pull request:
This PR adds settings for the Advanced tab of WP Super Cache. For the most part, the settings page is similar to the WordPress plugin version, with the following exceptions:

- The 'Caching On' and 'Caching Off' radio buttons have been consolidated into a single toggle switch.
- Checkboxes have been changed to toggle switches.
- Save button text was changed to 'Save Settings'.
- Extraneous text has not been carried over (e.g. the notes on garbage collection, Mod Rewrite Rules etc.).
- The UI better aligns with other Calypso settings pages.

[Calypso.live link](https://calypso.live/extensions/wp-super-cache?branch=add/wpsc-advanced-settings)

Below are screenshots of the individual sections. Note that hardcoded data is being used until the REST API is ready.

_Caching_

![caching](https://cloud.githubusercontent.com/assets/1190420/24309331/d59a5c7a-10a1-11e7-91a9-9aa082de1a54.jpg)

_Miscellaneous_

![miscellaneous](https://cloud.githubusercontent.com/assets/1190420/24309359/f24b906e-10a1-11e7-9e17-0fcb3db887b7.jpg)

_Advanced_

![advanced](https://cloud.githubusercontent.com/assets/1190420/24309369/fb3b51c8-10a1-11e7-9983-a01c3a648b5e.jpg)

_Cache Location_

![cache location](https://cloud.githubusercontent.com/assets/1190420/24309377/02184492-10a2-11e7-8054-ac77a9deb702.jpg)

_Expiry Time & Garbage Collection_

![expiry time](https://cloud.githubusercontent.com/assets/1190420/24309408/106d47b8-10a2-11e7-929d-7a3b483a1fd7.jpg)

_Accepted Filenames & Rejected URIs_

![accepted filenames](https://cloud.githubusercontent.com/assets/1190420/24309439/27335d20-10a2-11e7-9eb8-9e758b5d7412.jpg)

_Rejected User Agents_

![rejected user agents](https://cloud.githubusercontent.com/assets/1190420/24309451/347a552e-10a2-11e7-98c0-b29788deaf1e.jpg)

_Lock Down_

![lock down](https://cloud.githubusercontent.com/assets/1190420/24309456/39e0c4bc-10a2-11e7-8faa-23cabfe3bbe8.jpg)

_Directly Cached Files_

![directly cached files](https://cloud.githubusercontent.com/assets/1190420/24309472/46cf8fbe-10a2-11e7-926e-f2656089ad1c.jpg)

_Fix Configuration_

![fix configuration](https://cloud.githubusercontent.com/assets/1190420/24309476/4d6be1e2-10a2-11e7-87e4-feaa8cc2cee7.jpg)

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,180 @@
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
+					<option value="minutes_10">{ translate( 'Every 10 minutes' ) }</option>
+					<option value="minutes_30">{ translate( 'Every 30 minutes' ) }</option>
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( '30 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,180 @@
+					onChange={ handleSelect }
+					value={ fields.cache_schedule_interval || 'five_minutes_interval' }>
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
```
Hi! I've found a possible matching string that has already been translated **7** times:
`translate( '5 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,180 @@
+							disabled={ 'time' !== fields.cache_schedule_type }
+							onChange={ handleChange( 'cache_scheduled_time' ) }
+							value={ fields.cache_scheduled_time || '' } />
+						{ translate( 'HH:MM' ) }
```
Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'HH:mm', { context: 'momentjs format string (for LT)'} )` *ES Score: 22*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

yoavf commented:
We'll want to import the wpsc strings from [t.wp.org](https://translate.wordpress.org/projects/wp-plugins/wp-super-cache) once this lands. @donnapep  can you please add the " [Status] Needs i18n Review" label before merging? thanks!

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,213 @@
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
+					<option value="minutes_10">{ translate( 'Every 10 minutes' ) }</option>
+					<option value="minutes_30">{ translate( 'Every 30 minutes' ) }</option>
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( '30 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,213 @@
+					onChange={ handleSelect }
+					value={ cache_schedule_interval || 'five_minutes_interval' }>
+					<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+					<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
```
Hi! I've found a possible matching string that has already been translated **7** times:
`translate( '5 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,213 @@
+							disabled={ 'time' !== cache_schedule_type }
+							onChange={ handleChange( 'cache_scheduled_time' ) }
+							value={ cache_scheduled_time || '' } />
+						{ translate( 'HH:MM' ) }
```
Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'HH:mm', { context: 'momentjs format string (for LT)'} )` *ES Score: 22*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

donnapep commented:
@donnchawp I expect you'll be especially interested in [this bit](https://github.com/Automattic/wp-calypso/blob/e0eec8e49e17a4e0fc573b5676632b6bea4de753/client/extensions/wp-super-cache/components/wrap-settings-form.jsx#L60) where it lists out each section and which settings are used by those sections. I would imagine we'll need to discuss to ensure naming and structure are consistent.

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,208 @@
+						<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+						<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
+						<option value="minutes_10">{ translate( 'Every 10 minutes' ) }</option>
+						<option value="minutes_30">{ translate( 'Every 30 minutes' ) }</option>
```
Hi! I've found a possible matching string that has already been translated **28** times:
`translate( '30 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/extensions/wp-super-cache/components/advanced/expiry-time.jsx:
```diff
@@ -0,0 +1,208 @@
+						onChange={ handleSelect }
+						value={ cache_schedule_interval || 'five_minutes_interval' }>
+						<option value="five_minutes_interval">{ translate( 'Once every five minutes' ) }</option>
+						<option value="jetpack_sync_interval">{ translate( 'Every 5 minutes' ) }</option>
```
Hi! I've found a possible matching string that has already been translated **7** times:
`translate( '5 Minutes' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pgk commented on client/extensions/wp-super-cache/miscellaneous.jsx:
```diff
@@ -0,0 +1,190 @@
+	);
+};
+
+const getFormSettings = settings => {
```
I see this pattern in many of the files: a function we pass into `WrapSettingsForm` that returns a subset of settings fields. Was wondering if we should make this more declarative, by just passing the string array. 

Although passing in a function is more flexible.

donnapep replied:
Yeah, that would make sense to do as well. There's also [this](https://github.com/Automattic/wp-calypso/pull/12380/files#diff-f8410bc52a466929f7d47d48dd190e00R181) where more logic is needed to determine the contents of the array. It's conceivable that in the future other components might need to do something similar, hence your comment about flexibility. 😄

rickybanister commented:
This got merged before any kind of design review—I know it's not public-facing as yet, but should we take a run-through and open a big issue with any things we find? 

In following along I just noticed a few small bits like indentations of italic description text and whatnot.

donnapep commented:
@rickybanister I'm not sure we'll be doing design reviews in this early stage. We just want to get something in for now and then iterate later, which will likely completely change what we have. @dbtlr What are your thoughts on this?

dbtlr commented:
I'd say feel free to nitpick some of the details, but let's stay away from big refreshes for now. Hopefully, if we can collect enough of the smaller things into a task list, we can do it before this goes live as a final cleanup task. For now, though, the goal is to get this in there and functioning, so let's not get too caught up in fixing them just yet.

donnapep commented:
@rickybanister Given the above comments, how would you like to handle this going forward?

rickybanister commented:
I totally understand. I know we have to make progress here and just get things converted over, but I also want to be respectful to Calypso's contributing guidelines and design patterns. If we adhere to our own rules it will make it easier for other plugin developers to follow those as well. 

I'm happy to either open a single issue highlighting a few small things to adjust, or we can hop on a call and share the screen and just click through, pointing out/recording small discrepancies as we find them. Let me know which you prefer.

donnapep commented:
@rickybanister I'm going to be at a meetup this week, so if you could open an issue, that would be fantastic! There's also PR #12647 and #12689 if you'd like to have a look at those as well. Even if I don't get around to addressing the issues right away, at least I can avoid the same mistakes on the remaining tabs. Thx!
